Random Users/View Controllers/UsersTableViewController.swift

```diff
@@ -0,0 +1,149 @@
+        let userEmail = users[indexPath.row].email
+        
+        operationDict[userEmail]?.cancel()
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cellIdentifier = "userCell"
+        
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath) as? UserViewCell else {
+            NSLog("The dequeued cell is not an instance of UserViewCell")
+            
+            let errorCell = UserViewCell()
+            errorCell.userNameLabel.text = "Error occured while generating table cell"
+            return errorCell
+        }
+        
+        let user = users[indexPath.row]
+        
+        setUpCell(user: user, forCell: cell)
+        
+        return cell
+    }
+
+    // MARK: - Navigation
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        switch segue.identifier ?? "" {
+        case "showUserDetail":
+            guard let userDetailViewController = segue.destination as? UserDetailViewController else {
+                NSLog("Unexpected destination: \(segue.destination)")
+                return
+            }
+            
+            guard let selectedViewCell = sender as? UserViewCell else {
+                NSLog("Unexpected sender: \(sender ?? "No sender available")")
+                return
+            }
+            
+            guard let indexPath = tableView.indexPath(for: selectedViewCell) else {
+                NSLog("The selected cell is not being displayed by the table")
+                return
+            }
+            
+            let selectedUser = users[indexPath.row]
+            userDetailViewController.user = selectedUser
+        default:
+            NSLog("Unexpected segue indentifier: \(segue.identifier ?? "No segue available")")
+            return
+        }
+    }
+    
+    // MARK: - IBAction functions
+    @IBAction func getMoreUsersButtonPressed(_ sender: UIBarButtonItem) {
+        getUsers()
+    }
+    
+    // MARK: - Util functions
+    private func getUsers() {
+        controller.getUsersFromAPI { result in
+            do {
+                let userResults = try result.get()
+                DispatchQueue.main.async {
+                    self.users.append(contentsOf: userResults.results)
+                }
+            } catch {
+                NSLog("Error fetching users. Logging function called from UsersTableViewController")
+                return
+            }
+        }
+    }
+    
+    private func asyncReload() {
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+        }
+    }
+    
+    private func setUpCell(user: User, forCell cell: UserViewCell) {
+        cell.user = user
+        
+        if let cachedThumbnailData = thumbnailCache.getValue(for: user.email) {
+            cell.thumbnail = UIImage(data: cachedThumbnailData)
+        }
```

Comment:
Should there be a return out of the func?  Why do the rest if the image is already cached?